sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpPipelinePolicies.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.policy;
+
+import com.azure.core.http.HttpPipelinePosition;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a set of {@link HttpPipelinePolicy HttpPipelinePolicies}.
```

Comment:
Talk about how it will allow for getting the list based on their positioning. Also, consider an API that allows retrieval of the whole list in an unmodifiable collection.

Reply:
Return an unmodifiable collection or a new collection? Both would work in preventing the internal collection from being mutated but the latter would make it easier to make changes and create a new `HttpPipelinePolicies`.